### PR TITLE
code clean up related to error checking and forced operations in state.

### PR DESCRIPTION
**Pull request by hmlanigan**
## Description of change

For Operations in state using ForcedOperation, add FatalError() and use to replace oft replicated error checking sequence where possible.

## QA steps

Should have no impact to juju operations for tests.

### CODE REVIEW

File: state/forcedoperation.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
+// Copyright 2019 Canonical Ltd.
```
Comment (hmlanigan) [inline state/forcedoperation.go]: This code could also move to modeloperation.go?  Does not belong in unit.go though.

File: state/forcedoperation.go [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+// AddError adds an error to the collection of errors for this operation.
+func (op *ForcedOperation) AddError(one ...error) {
+	op.Errors = append(op.Errors, one...)
+}
+
+// FatalError returns true if the err is not nil and Force is false.
+// If the error is not nil, it's added to the slice of errors for the
+// operation.
+func (op *ForcedOperation) FatalError(err error) bool {
```
Comment (hmlanigan) [inline state/forcedoperation.go]: Is there a better name?

Comment (manadart) [reply state/forcedoperation.go]: `CheckFatal` comes to mind, but I think this is OK.

File: state/forcedoperation.go (hunk 1 above)
Comment (manadart) [reply state/forcedoperation.go]: I think it is better here than modeloperation.go.

**Comment by hmlanigan** [comment]
$$merge$$

**Comment by hmlanigan** [comment]
interesting, i can only reproduce the issue with 2.5, not 2.6 nor develop.  was looking at qa for this pr.

**Comment by hmlanigan** [comment]
$$merge$$

**Pull request merged**